### PR TITLE
CIRC-723 Overdue fine not calculated properly when closed time not counted

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose 
Implement logic for calculating overdue fine when closed time is not counted.

Resolves [CIRC-723](https://issues.folio.org/browse/CIRC-723)

## Approach
- update OverduePeriodCalculatorService;
- update unit tests;
- implement integrated tests;

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/test/java/api/loans/CheckInByBarcodeTests.java [hunk 1]
```diff
@@ -919,6 +924,78 @@ public void noOverdueFineShouldBeChargedForOverdueFinePolicyWithNoOverdueFine()
     assertThat("Fee/fine action record shouldn't be created", createdFeeFineActions, empty());
   }
 
+  @Test
+  public void overdueFineShouldBeChargedWhenItemIsOverdueForSevenHours() {
```
Comment (alexanderkurash) [inline src/test/java/api/loans/CheckInByBarcodeTests.java:928]: I think method name is misleading

Comment (marcjohnson-kint) [reply src/test/java/api/loans/CheckInByBarcodeTests.java:928]: @alexanderkurash Please could you expand upon your feedback. Why do you think it is misleading? What might you suggest as an alternative?

Comment (alexanderkurash) [reply src/test/java/api/loans/CheckInByBarcodeTests.java:928]: @marcjohnson-kint I don't think it reflects the purpose of the test. It looks like we're trying to check that hourly policies are working correctly, not that we charge a fine when item is exactly 7hrs overdue. I would probably go with something like overdueFineCalculatedCorrectlyWhenHourlyFeeFinePolicyIsApplied

Comment (roman-barannyk) [reply src/test/java/api/loans/CheckInByBarcodeTests.java]: renamed

**Pull request merged**